rs/consensus/src/dkg/payload_builder.rs

```diff
@@ -308,18 +302,11 @@ fn compute_remote_dkg_data(
         let mut expected_configs = Vec::new();
         for config in low_high_threshold_configs {
             let dkg_id = config.dkg_id();
-            // Check if we have a transcript in the previous summary for this config, and
-            // if we do, move it to the new summary.
-            if let Some((id, transcript)) = previous_transcripts
-                .iter()
-                .find(|(id, _)| eq_sans_height(id, dkg_id))
-            {
-                new_transcripts.insert(id.clone(), transcript.clone());
-            }
```

Comment:
I think another reason for this code could be because we need to reshare two transcripts (high & low) to respond to the reshare request. So if we only manage to create one of them in a single interval, then we can work on the second transcript in the next interval, and don't have to restart both of them from scratch.

Similarly, if we delivered both transcripts in the first summary block, but by the time we create the second summary block still haven't executed the first one, then we would also redo the resharing.

Reply:
I see. I think the second reason might be the decisive one. If we have a Finalization - Certification Gap longer than the DKG interval, we will start creating a new transcript, which consumes resources and will then eventually be delivered to execution twice. 

Its a very theoretical edge-case though, so I don't know what would be the correct way to proceed. Do you think we should keep it?

Reply:
We added a hard cap of 70 heights on the notarization - certification gap some time ago, so it shouldn't be possible assuming the DKG interval is longer. But I'm not sure if that's something we should rely on. How much does not having this case simplify the other changes you have planned?

Reply:
It's not really so closely related to the optimization. I'm just trying to reduce unnecessary complexity. 

I still don't fully understand why this case was added. Even if we just copy the data over, we would deliver the response twice (just not recompute it twice). The correct way to handle this would be to look back into the chain (until the last CUP height, which would be more than 1 DKG intervall in this case) and see if we already have a transcript, which will eventually get delivered and make the context go away.

I'm also ok with leaving the case in, I just think this is weird, as nobody understands it, it is not covered by tests, and it was likely never triggered on mainnet.

Reply:
As discussed, we will not remove this at the time being.
We couldn't fully reconstruct the purpose of this case, and will add a log message instead.

The whole dkg submodule needs a make-over at some not so distant point in the future. We willl check by then, what to do with this case